distributed/scheduler.py

```diff
@@ -3241,6 +3237,17 @@ def worker_objective(self, ts: TaskState, ws: WorkerState) -> tuple:
         else:
             return (start_time, ws.nbytes)
 
+    def worker_objective_no_deps(self, ws: WorkerState) -> tuple[float, float]:
+        """
+        Objective function to determine the least-busy worker.
+
+        Meant for use with tasks where dependencies are not relevant for scheduling
+        (no dependencies, or widely-shared).
+
+        Minimize worker occupancy. If a tie then break with data storage.
+        """
+        return (ws.occupancy / ws.nthreads, ws.nbytes / (ws.memory_limit or 1))
```

Comment:
Why do we need a different objective function? Without dependencies the original reduces to almost the same thing. It prioritites by

`(nactors, start_time, nbytes)`

i.e.
`(nactors, ws.occupancy / ws.nthreads, nbytes)`

which is the same thing as you are proposing here. With the exception that you're dividing nbytes by the memory limit.

Looks like we should just use the same objective function. I suggest not dividing for now (and do this / discuss this in another PR)

Reply:
This one doesn't iterate over dependencies. The other does. By the definition of `is_rootish`, it's possible for there to be dependencies. But the point of root-ish-ness is that we should be ignoring them.